index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Compiles basic twig templates for patternlab iframe wrapper
+ *
+ * USAGE:
+ * var patternlabWrapper = require('patternlab-wrapper');
+ *
+ * patternlabWrapper({
+ *   dest: './app/public',
+ *   patternPaths: {"atoms":{"colors":"00-atoms-01-global-00-colors","redirect":"00-atoms-01-global-01-redirect",...}}
+ *   patternNavHTML: '<ul class="patterns-menu sg-nav">
+    <li class="sg-nav-"><a class="sg-acc-handle">base</a>
+        <ul class="sg-acc-panel">
+
+            <li class="sg-nav-"><a class="sg-pop" href="app/bower_components/pattern-library/patterns/base/blockquote">Blockquote</a></li>
+            <li class="sg-nav-"><a class="sg-pop" href="app/bower_components/pattern-library/patterns/base/body">Body</a></li>
+        </ul>
+    </li>
+    </ul>'
+ * });
+ *
+ */
+
+/**
+ *
+ * @type {Twig.exports|exports}
+ */
+
+var Twig = require('twig');
+var fs = require('fs');
+var fsextra = require('fs-extra');
+
+var patternlabFiles = [
+  'styleguide', 'js', 'images', 'fonts', 'data'
+];
+
+/**
+ *
+ * @type {{dest: string, patternNav: string, patternNavTemplate: string, ishControls: string, ishControlsTemplate: string, patternPaths: {base: Array}, src: string}}
+ */
+var optionsDefault = {
+  dest: './public',
+  patternNavHTML: '',
+  patternNavTemplate: './_patternlab-files/partials/patternNav.twig',
+  ishControlsHTML: '',
+  ishControlsTemplate: './_patternlab-files/partials/ishControls.twig',
+  patternPaths: {base: []},
+  src: './_patternlab-files'
+};
+
+function patternlabWrapper(options) {
+  options = options || {};
+  options.patternNavHTML = options.patternNavHTML || Twig.twig({data: fs.readFileSync(optionsDefault.patternNavTemplate, 'utf8')}).render();
+  options.ishControlsHTML = options.ishControlsHTML || Twig.twig({data: fs.readFileSync(optionsDefault.ishControlsTemplate, 'utf8')}).render();
+  options.patternPaths = options.patternPaths || optionsDefault.patternPaths;
+  options.dest = options.dest || optionsDefault.dest;
+  options.src = options.src || optionsDefault.src;
+
+  var indexTemplate = Twig.twig({data: fs.readFileSync('./_patternlab-files/index.twig', 'utf8')});
+
+  fs.writeFile(options.dest + '/index.html', indexTemplate.render({
```

Comment:
it can be done by:
```js
var Q  = require('q');
var fs  = require('fs');
var writefile = function (fname, enc) {
    return Q.promise(function(resolve, reject){
        fs.writefile(fname, enc, function (error, text) {
            if (error) {
                // 'error' is already an error object, you don't need 'new Error()'.
                reject(error);
            } else {
                resolve(text);
            }
        });
    });
};
```

But I'm not sure if we need them here, callback "just works".